src/rgw/cls_fifo_legacy.cc

```diff
@@ -1084,7 +1086,7 @@ int FIFO::trim(std::string_view markstr, bool exclusive, lr::AioCompletion* c) {
   const auto pn = info.tail_part_num;
   const auto part_oid = info.part_oid(pn);
   l.unlock();
-  auto trimmer = new Trimmer(this, marker->num, marker->ofs, pn, exclusive, c);
+  std::unique_ptr<Trimmer> trimmer(new Trimmer(this, marker->num, marker->ofs, pn, exclusive, c));
   ++trimmer->pn;
```

Comment:
This doesn't seem right unless we add a release at some point. We still want Trimmer to hang around until the asynchronous operations finish, this would free it as soon as the function exits, I believe.

Reply:
so where is the trimmer actually deleted?

Reply:
ok. i see that the when the call is successful, the deletion is done in: `FIFO::update_callback()`
will revert the change (and rename the commit...)

Reply:
two thoughts about this one:
* if it isn't immediately obvious where this memory is released, the allocation probably deserves a clarifying comment
* it may still be a good idea to use `unique_ptr` within this function for exception-safety, with a `trimmer.release()` in the successful case

Reply:
> * it may still be a good idea to use `unique_ptr` within this function for exception-safety, with a `trimmer.release()` in the successful case

this is actually worse in other places in the file. e.g. in the: `FIFO::update_callback()`
probably worth fixing in all places.
the "deletion logic" is quite complex, and changing that to "unique_ptr release logic" going to be simpler

Reply:
lets do that in a separate PR.
(tracked here: https://tracker.ceph.com/issues/47848)

Reply:
lets leave that to another PR.
tracked here: https://tracker.ceph.com/issues/47848